### PR TITLE
build/configs: Configure RT IDE debugging with esp32 boards

### PR DESCRIPTION
**Pull request by zhouxinhe**
- ESP-WROVER-KIT boards have built-in JTAG functionality, IDE debugging
  is ready to use.

- ESP32-DevKitC boards have not built-in JTAG adapter, user need to find
  one and also provide a configuration file.
  See build/configs/esp32_DevKitC/scripts/esp32.cfg for more details.

- Additional settings in RT IDE (will be improved in near future):
  Debug Configurations - GDB OpenOCD Debugging
  - Main
    * C/C++ Application: build/output/bin/tinyara.elf
  - Debugger
    * GDB Client Setup
     Executable: /.../tizen-studio/rt-ide/tools/bin/xtensa-esp32-elf-gdb (if not exist, copy it from toolchain)

- Add `How to use GDB` description in README.md files for esp_wrover_kit and esp32_DevKitC.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [57bdb14e8444fbbfbd1f62348a1295cdbf58741b] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [57bdb14e8444fbbfbd1f62348a1295cdbf58741b] - Code Rule Check (C++) OK.

**Comment by zhouxinhe** [comment]
This PR is to fix issue #4029

File: build/configs/esp32_DevKitC/scripts/target/esp_common.cfg [hunk 1]
```diff
@@ -0,0 +1,116 @@
+# Common ESP chips definitions
+
+if { [info exists ESP_RTOS] } {
+	set _RTOS "$ESP_RTOS"
+} else {
+	set _RTOS "FreeRTOS"
```
Comment (xixidodo) [inline build/configs/esp32_DevKitC/scripts/target/esp_common.cfg]: do we need to remove those RTOS config?

Comment (zhouxinhe) [reply build/configs/esp32_DevKitC/scripts/target/esp_common.cfg]: I bring this file from openocd-esp32 without any change, the RTOS debugging supports FreeRTOS only, so I put `set ESP_RTOS none` in upper esp32.cfg file.

**Comment by seinfra** [comment]
Target : [4e2d845f0a5eac47dfebe42e0c01120618d4072d] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [4e2d845f0a5eac47dfebe42e0c01120618d4072d] - Code Rule Check (C++) OK.

**Comment by zhouxinhe** [comment]
@sunghan-chang 
Added the description of debugging in README files.
Please help to review, thanks.

**Pull request merged**